--- a/dwm.c	2012-03-28 22:57:19.189221851 +0300
+++ b/dwm.c	2012-03-28 22:57:19.191221853 +0300
@@ -1409,7 +1409,11 @@ resizeclient(Client *c, int x, int y, in
 	c->oldy = c->y; c->y = wc.y = y;
 	c->oldw = c->w; c->w = wc.width = w;
 	c->oldh = c->h; c->h = wc.height = h;
-	wc.border_width = c->bw;
+	wc.border_width = c->mon->lt[c->mon->sellt]->arrange == monocle && !c->isfloating ? 0 : c->bw;
+	if(c->mon->lt[c->mon->sellt]->arrange == monocle && !c->isfloating) {
+		c->w = wc.width += 2 * c->bw;
+		c->h = wc.height += 2 * c->bw;
+	}
 	XConfigureWindow(dpy, c->win, CWX|CWY|CWWidth|CWHeight|CWBorderWidth, &wc);
 	configure(c);
 	XSync(dpy, False);
