--- a/dwm.c	2012-03-28 22:57:19.198221851 +0300
+++ b/dwm.c	2012-03-28 22:57:19.201221853 +0300
@@ -866,6 +866,8 @@ drawtext(const char *text, unsigned long
 
 void
 enternotify(XEvent *e) {
+	if(clicktofocus) return;
+
 	Client *c;
 	Monitor *m;
 	XCrossingEvent *ev = &e->xcrossing;
@@ -1255,6 +1257,8 @@ monocle(Monitor *m) {
 
 void
 motionnotify(XEvent *e) {
+	if(clicktofocus) return;
+
 	static Monitor *mon = NULL;
 	Monitor *m;
 	XMotionEvent *ev = &e->xmotion;
